### ISSUE TITLE
Discussion: moving to alternative git host

### ISSUE THREAD

**Issue by xXxEMBERGHOSTSQUADDIExXx** [d3athrow/vgstation13#35422]
With the nuking of pedr69 over some very tame language its clear that the previous attempts at appeasement viz. the sacrifice of server culture has done nothing to make github any less eager to jump at the opportunity to attack us

If github can on a whim delete all of someones prs, issues, etc then this is harmful to the project

one thing to keep in mind is that this isn't necessarily github doing this unbidden; certain users have been known to weaponize this tendency of github's to attack other users or the repo itself

I suggest we move to https://gitgud.io, a free speech supporting variant of gitlab. The tooling is better and better than it has been.

Or alternatively we could use a self hosted https://gitea.com/ instance.

note that CI tools exist for both of these options:
gitgud.io:

> Kanban boards, Groups, Private repos, GitLab Pages, CI Pipelines, Community Issue and Feedback on private repos, and more
> Free shared CI. Just request access from us via a communication channel
> 

gitea:

> Gitea features an integrated CI/CD system, Gitea Actions, that is compatible with GitHub Actions. Users can create workflows using the familiar YAML format or utilize numerous existing plugins. Additionally, a wide variety of third-party CI/CD integrations are available.

something to consider is that there doesn't necessarily need to be a "hard switch" moment. there could be two mirrored repos that are synced with each other, so that prs could be applied to both platforms for the switching period. 

discuss

**Comment by Eneocho** [comment]
PEDR I WARNED YOU
I WARNED YOU ABOUT STAIRS BRO
I would support moving to gitgud. Stop all new PRs except bugfixes, accept or close the current PRs. Once we have 0 PRs pending, we fully move to gitgud, whilst keeping this as an archived version.
Not sure if we could import all the issues, but we can manage.
Also, it's less that github deletes all your stuff, but rather than it hides all your content for everyone that isn't you until you delete the offending post(s) and apologize; unless you were a particularly egregious gamer so they hide it forever and suspend your account (effectively deleting it).

**Comment by boy2mantwicethefam** [comment]
If it is self-hosted we could tie accounts to ckeys and have more control over voting to prevent situations like permabanned players or alt-accounts voting.

**Comment by xXxEMBERGHOSTSQUADDIExXx** [comment]
@Eneocho actually as long as there weren't conflicts the server could pull from both gitgud and github at the same time, so the swapover period could be a little smoother that way. just disable making new prs on github and only accept new ones on gitgud so its a nice gradual switch

**Comment by DorainGrey** [comment]
Even though mapdiff bot is broke at the moment, the CI tools are quite powerful and I don't really want to lose them.
I just think at the end of the day, it's really not worth it. The github isn't a social forum, despite some people terminally dwelling here. 
Making it harder to contribute & losing a lot of history for the sake of a few people wanting to say the gamerword on the code repo, is a bad tradeoff. We need less barriers to coding & more content not more.


Certainly if we were to consider this, both gitea and gitgud are both bad options IMO.

Gitlab is easy to self host, has a lot of powerful features and is open source. Definitely the least transition pain & most security of usage. IE (It won't shutdown like gitgud and since it's enterprise & free versions are popular won't die)

The one thing I do like is preventing alting, but I think this is fairly clear even on github.

**Comment by xXxEMBERGHOSTSQUADDIExXx** [comment]
CI tools exist for both options. i updated the OP to reference this (its been the main cited reason why not to do this in past discussions)

**Comment by DorainGrey** [comment]
Yes they exist, but are you committing to migrating them to the different hosts.
At the moment they are plug and play.

Mirroring isn't necessarily a bad idea, this was done for a long long time by Nexis on the gitlab, I don't know if you can sync PRs and stuff though.

**Comment by ghost** [comment]
+1, but who would actually be willing to do this? None of the people legitimately concerned are in control.
